### PR TITLE
fix: reorder electron launching sequence to eliminate timing issue

### PR DESCRIPTION
**Pull request by ryanthemanuel**
### User facing changelog

n/a

### Additional details

Due to some changes in how we transition tests made on #19915 we are seeing a race condition with electron where the cdp automation is not set up in time to handle the events that get triggered from running tests. This PR will ensure that we have CDP loaded before we navigate to the new url (incidentally this is how chrome already works).

### PR Tasks

- [X] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/server/lib/browsers/electron.js [hunk 1]
```diff
@@ -198,8 +196,6 @@ module.exports = {
     }
 
     return this._launch(win, url, automation, preferences)
-    .tap(_maybeRecordVideo(win.webContents, preferences))
```
Comment (lmiller1990) [inline packages/server/lib/browsers/electron.js:201]: Good to see less async callbacks using `tap`. I'm sure it's an antipattern to use `tap` for anything other than logging, as noted by the first comment on the `tap` docs: http://bluebirdjs.com/docs/api/tap.html

I am pretty sure `tap` will never wait, and previously `maybeRecordVideo` was an async function, so this was basically a race condition waiting to happen.

File: packages/server/test/unit/browsers/electron_spec.js [hunk 2]
```diff
@@ -290,6 +292,34 @@ describe('lib/browsers/electron', () => {
         })
       })
     })
+
+    it('registers onRequest automation middleware and calls show when requesting to be focused', function () {
```
Comment (lmiller1990) [inline packages/server/test/unit/browsers/electron_spec.js:296]: Seems fine but any reason to move this chunk of code?

**Review by lmiller1990 — APPROVED** [review]
Changes seem fine, this would have been really hard to track down (assuming this fixes it). Left one question but not really a blocker.

**Comment by ryanthemanuel** [comment]
> Changes seem fine, this would have been really hard to track down (assuming this fixes it). Left one question but not really a blocker.
> 
> I wonder if we can somehow have a public API that all browsers are forced to comply to for consistency, like:
> 
> ```
> interface BrowserAutomationLifecycle {
> 
>   connectAutomation: Function...
>   maybeRecord: Function...
> }
> ```
> 
> That would enforce all our browsers to implement the same methods, so we would avoid a "it works differently in X browser". Then we'd have some common place where all the methods are called.
> 
> I'm not sure sure if this even makes sense, just a thought I had when reading "this is how Chrome already does it". My first thought was "why are the different!".

I definitely agree with you. As I was digging into #19915, I noticed several areas that could be refactored. Some I did, and some I scoped out for the sake of time

File: packages/server/lib/browsers/electron.js [hunk 3]
```diff
@@ -263,14 +259,16 @@ module.exports = {
       )
     })
     .then(() => {
-      return win.loadURL(url)
+      return win.loadURL('about:blank')
     })
+    .then(() => this._getAutomation(win, options, automation).then((cdpAutomation) => automation.use(cdpAutomation)))
```
Comment (emilyrohrbough) [inline packages/server/lib/browsers/electron.js:264]:
```suggestion
    .then(() => this._getAutomation(win, options, automation)
    .then((cdpAutomation) => automation.use(cdpAutomation)))
```

File: packages/server/lib/browsers/electron.js [hunk 4]
```diff
@@ -263,14 +259,16 @@ module.exports = {
...
+    .then(() => Promise.all([_maybeRecordVideo(win.webContents, options), this._handleDownloads(win, options.downloadsFolder, automation)]))
```
Comment (emilyrohrbough) [inline packages/server/lib/browsers/electron.js:265]:
```suggestion
    .then(() => Promise.all([
      _maybeRecordVideo(win.webContents, options),
      this._handleDownloads(win, options.downloadsFolder, automation)
    ]))
```

**Review by emilyrohrbough — APPROVED** [review]

**Review by emilyrohrbough — DISMISSED** [review]

**Review by ZachJW34 — APPROVED** [review]

**Pull request merged**